tests/generator/test_errors.py

```diff
@@ -999,3 +999,268 @@ def test_nm_only_supports_unicast_routes(self):
+  version: 2
+  ethernets:
+    eth0:
+      dhcp4: true
+    eth1:
+      dhcp4: yesplease
+    eth2:
+      renderer: NetworkManager
+      dhcp4: false''', expect_fail=False, skip_generated_yaml_validation=True, ignore_errors=True)
+
+        self.assertTrue(self.file_exists('10-netplan-eth0.network'))
+        self.assertTrue(self.file_exists('10-netplan-eth1.network'))
+        self.assertTrue(self.file_exists('netplan-eth2.nmconnection', backend='NetworkManager'))
+
+        self.assertIn('Skipping definition due to parsing errors. eth1', out)
+
+    def test_ignore_errors_multiple_files(self):
+        ''' Test that a bad YAML file will be ignored '''
+        out = self.generate('''network:
+  version: 2
+  ethernets:
+    eth0:
+      dhcp4: true''', confs={'b': '''network:
+  ethernets:
+    eth1: {}''',
+                             'c': ''':''',
+                             'd': '''network:
+  ethernets:
+    eth2: {}'''}, expect_fail=False, skip_generated_yaml_validation=True, ignore_errors=True)
+
+        self.assertTrue(self.file_exists('10-netplan-eth0.network'))
+        self.assertTrue(self.file_exists('10-netplan-eth1.network'))
+        self.assertTrue(self.file_exists('10-netplan-eth2.network'))
+        self.assertIn('Skipping YAML file due to parsing errors.', out)
+
+    def test_ignore_syntax_errors(self):
+        ''' Test that an error in the same netdef in the next file will not remove the netdef '''
+        out = self.generate('''network:
+  version: 2
+  ethernets:
+    eth0:
+      dhcp4: true''', confs={'b': '''network:
+  ethernets:
+    eth0:
+      abc: 123'''}, expect_fail=False, skip_generated_yaml_validation=True, ignore_errors=True)
+
+        self.assertTrue(self.file_exists('10-netplan-eth0.network'))
+        self.assertIn('Skipping definition due to parsing errors. eth0', out)
+
+    def test_ignore_errors_dependencies(self):
+        ''' Test that an interface that depends on a bad interface will still have configuration generated '''
+        out = self.generate('''network:
+  version: 2
+  ethernets:
+    eth123:
+      dhcp4: true
+    eth321:
+      dhcp4: 1''', confs={'b': '''network:
+  bridges:
+    br0:
+      interfaces: [ eth123 ]
+    br1:
+      interfaces: [ eth321 ]'''}, expect_fail=False, skip_generated_yaml_validation=True, ignore_errors=True)
+
+        self.assertTrue(self.file_exists('10-netplan-eth123.network'))
+        self.assertTrue(self.file_exists('10-netplan-br0.network'))
+        self.assertFalse(self.file_exists('10-netplan-321.network'))
+        self.assertTrue(self.file_exists('10-netplan-br1.network'))
+        self.assertIn('Skipping definition due to parsing errors. eth321', out)
+
+    def test_ignore_errors_bad_bond(self):
+        self.generate('''network:
+  version: 2
+  ethernets:
+    eth0: {}
+  bonds:
+    bond0:
+      interfaces: [eth0]
+      badkey: badvalue''', expect_fail=False, skip_generated_yaml_validation=True, ignore_errors=True)
```

Comment:
question: I assume this would still be fine if we define `badkey: badvalue` first, before `interfaces: [...]`?

Reply:
If the error happened before `interfaces: [eth0]` this fields wouldn't be parsed. The `bond0` would still be there, but with an empty list of members.

Reply:
Thanks for the explanation, this should be fine!